### PR TITLE
Add volume mixer logic

### PR DESCRIPTION
**Pull request by mjschuetze102**
Adds the logic on the backend that will handle storing the volume levels for each application.

### CODE REVIEW

File: VolumeLogic/IObserver.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import abc
+from VolumeLogic import Subject
+
+
+class IObserver(metaclass=abc.ABCMeta):
+    """Creates interface for classes that need to be notified of a change."""
+
+    @classmethod
+    def __subclasshook__(cls, subclass):
+        return (hasattr(subclass, 'update') and
```
Comment (Michael-Longley) [inline VolumeLogic/IObserver.py:10]: Will hasattr return true if the method raises `NotImplementedError`?

File: VolumeLogic/VolumeMixer.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    """Collection of Applications whose volume is being controlled."""
+
+    actions = {
+        ChangedValue.VOLUME: lambda application, value: application.volume(value),
+        ChangedValue.COLOR: lambda application, value: application.color_matrix(value),
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.applications = [None] * 5
```
Comment (Michael-Longley) [inline VolumeLogic/VolumeMixer.py]: Can we get rid of the magic number here? set a global 5 which a descriptive name and use that.

**Review by Michael-Longley — CHANGES_REQUESTED** [review]
Check the question and remove the magic number and LGTM

File: VolumeLogic/IObserver.py (hunk 1 above)
Comment (mjschuetze102) [reply VolumeLogic/IObserver.py:10]: Classes that inherit from IObserver don't actually need to have an update method in normal python. With abc, the call to \_\_subclasshook__ is called to make sure that the update method is included in the subclass and that it is implemented.

**Review by Michael-Longley — APPROVED** [review]

**Pull request merged**